torch/ao/quantization/experimental/observer.py

```diff
@@ -111,20 +113,39 @@ def _calculate_qparams(
 
         return (self.gamma, quantization_levels, level_indices)
 
+    def float_to_apot(self, x, levels, indices):
+        levels_lst = list(levels)
+        indices_lst = list(indices)
+
+        min_delta = math.inf
+        best_idx = 0
+
+        for level, idx in zip(levels_lst, indices_lst):
+            cur_delta = abs(level - x)
+            if cur_delta < min_delta:
+                min_delta = cur_delta
+                best_idx = idx
+
+        return best_idx
+
+    def apot_to_float(self, x_apot, levels, indices):
+        idx = list(indices).index(x_apot)
+        return levels[idx]
+
 class APoTObserver(NonUniformQuantizationObserverBase):
     def __init__(
         self,
-        min_val=torch.Tensor,
-        max_val=torch.Tensor,
+        min_val=None,
+        max_val=None,
         b=0,
             k=0) -> None:
         super(APoTObserver, self).__init__(min_val, max_val, b, k)
 
     def calculate_qparams(self, signed):
-        return self._calculate_qparams(self.min_val, self.max_val, signed)
+        return self._calculate_qparams(signed, self.min_val, self.max_val)
 
-    def _calculate_qparams(self, min_val, max_val, signed):
-        return super(APoTObserver, self)._calculate_qparams(min_val, max_val, signed)
+    def _calculate_qparams(self, signed, min_val, max_val):
+        return super(APoTObserver, self)._calculate_qparams(signed, min_val, max_val)
```

Comment:
why there is a standard way for NonUniformQuantizationObserverBase to calculate qprams?

Reply:
yep this should be APoT calculate qparams